src/visualization/makie_visualization.jl

```diff
@@ -0,0 +1,263 @@
+"""
+function compute_triangle_area(tri)
+    A,B,C = tri
+    return .5*(A[1]*(B[2] - C[2]) + B[1]*(C[2]-A[2]) + C[1]*(A[2]-B[2]))
+end
+
+"""
+  function plotting_triangulation(reference_plotting_coordinates)
+
+Computes a triangulation of the points in `reference_plotting_coordinates`, which is a tuple containing
+vectors of plotting points on the reference element (e.g., reference_plotting_coordinates = (r,s)).
+The reference element is assumed to be [-1,1]^d.
+
+This function returns `t` which is a `3 x N_tri` Matrix{Int} containing indices of triangles in the
+triangulation of the plotting points, with zero-volume triangles removed.
+
+For example, r[t[1, i]] returns the first reference coordinate of the 1st point on the ith triangle.
+"""
+function plotting_triangulation(reference_plotting_coordinates,tol=50*eps())
+
+    # on-the-fly triangulation of plotting nodes on the reference element
+    triin = Triangulate.TriangulateIO()
+    triin.pointlist = permutedims(hcat(reference_plotting_coordinates...))
+    triout,_ = Triangulate.triangulate("Q", triin)
+    t = triout.trianglelist
+
+    # filter out sliver triangles
+    has_volume = fill(true, size(t,2))
+    for i in axes(t, 2)
+        ids = @view t[:, i]
+        x_points = @view triout.pointlist[1, ids]
+        y_points = @view triout.pointlist[2, ids]
+        area = compute_triangle_area(zip(x_points, y_points))
+        if abs(area) < tol
+            has_volume[i] = false
+        end
+    end
+    return t[:, findall(has_volume)]
+end
+
+"""
+  function plotting_interpolation_matrix(dg; kwargs...)
+
+Interpolation matrix which maps discretization nodes to a set of plotting nodes.
+Defaults to the identity matrix of size `length(solver.basis.nodes)`, and interpolates
+to equispaced nodes for DGSEM (set by kwarg `nvisnodes` in the plotting function).
+
+Example:
+```julia
+A = plotting_interpolation_matrix(dg)
+A * dg.basis.nodes # => vector of nodes at which to plot the solution
+```
+"""
+# Note: we cannot use UniformScaling to define the interpolation matrix since we use it with `kron`
+# to define a multi-dimensional interpolation matrix later.
+plotting_interpolation_matrix(dg; kwargs...) = I(length(dg.basis.nodes))
+
+function plotting_interpolation_matrix(dg::DGSEM; nvisnodes = 2*length(dg.basis.nodes))
+    return polynomial_interpolation_matrix(dg.basis.nodes, LinRange(-1, 1, nvisnodes))
+end
+
+"""
+  function generate_plotting_triangulation(sol::TrixiODESolution, variable_to_plot::Int; nvisnodes=5)
+
+Generates a plotting triangulation which can be used by Makie's `mesh` function.
+
+Arguments:
+- sol: TrixiODESolution
+- variable_to_plot: index of solution field to plot
+
+Example:
+```julia
+tri = generate_plotting_triangulation(sol,1)
+Makie.mesh(tri,color=getindex.(tri.position,3))
+```
+The color is specified to be the z-value of the solution, which there does not seem to be a default solution for yet.
+"""
```

Comment:
Same as above: If it's an internal function, the docstring should be converted into a comment